app/assets/javascripts/discourse/app/mixins/composer-upload-uppy.js

```diff
@@ -343,6 +357,107 @@ export default Mixin.create({
     });
   },
 
+  _useS3MultipartUploads() {
+    const headers = {
+      "X-CSRF-Token": this.session.csrfToken,
+      "Content-Type": "application/json",
+    };
+    const self = this;
+
+    this._uppyInstance.use(AwsS3Multipart, {
+      // controls how many simultaneous _chunks_ are uploaded, not files,
+      // which in turn controls the minimum number of chunks presigned
+      // in each batch (limit / 2)
+      //
+      // the default, and minimum, chunk size is 5mb. we can control the
+      // chunk size via getChunkSize(file), so we may want to increase
+      // the chunk size for larger files
+      limit: 10,
+
+      createMultipartUpload(file) {
+        return ajax(getURL("/uploads/create-multipart.json"), {
+          type: "POST",
+          headers,
+          data: JSON.stringify({
```

Comment:
I'm curious why we need to convert the object to a json string here? I think the ajax library already handles that for us.

Reply:
I was able to change all these by getting rid of `headers`, which was not needed, except for `complete-multipart` which has an array in the data, so I had to explicitly stringify and set the content type to JSON, otherwise rails put the params into a weird format I couldn't call `map` on in the controller